### PR TITLE
tikv: modify TiKV & PD config docs for TiKV API v2

### PR DESCRIPTION
**Pull request by pingyu**
Signed-off-by: pingyu <yuping@pingcap.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Modify TiKV & PD configuration docs for TiKV API v2 feature.
See [rendered](https://github.com/pingyu/pingcap-docs-cn/blob/tikv-api-v2-v640/tikv-configuration-file.md#causal-ts-%E4%BB%8E-v610-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5).

Benchmark of TiKV default config refers to [here](https://clinic.pingcap.com.cn/portal/#/orgs/31/clusters/7153163782686330475?from=1665475676&to=1665480449).
Benchmark of `tso-update-physical-interval` refers to https://github.com/tikv/pd/pull/5437.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by pingyu** [comment]
@haojinming @zeminzhou PTAL, thanks~

**Review by haojinming — APPROVED** [review]
LGTM

**Review by zeminzhou — APPROVED** [review]
LGTM~

**Comment by pingyu** [comment]
@sunxiaoguang @haojinming @zeminzhou 
The `available-interval` is renamed to `alloc-ahead-buffer` in https://github.com/tikv/tikv/pull/13597.
PTAL, thanks~

**Review by sunxiaoguang — APPROVED** [review]
LGTM

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: pd-configuration-file.md [hunk 1]
```diff
@@ -92,6 +92,14 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
 + 强制让该 PD 以一个新集群启动，且修改 raft 成员数为 1。
 + 默认值：false
 
+### `tso-update-physical-interval` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
++ TSO 物理时钟更新周期。
++ 在一个 TSO 物理时钟更新周期内，PD 最多提供 262144 个 TSO。如果需要更多的 TSO，可以将这个参数调小。最小值为 `1ms`。
```
Comment (Oreoxmt) [inline pd-configuration-file.md:98]:
```suggestion
+ 在默认的一个 TSO 物理时钟更新周期内 (50ms)，PD 提供 262144 个 TSO。如果需要更多的 TSO，可以将这个参数调小。
```

File: pd-configuration-file.md [hunk 2]
```diff
@@ -92,6 +92,14 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
 + 强制让该 PD 以一个新集群启动，且修改 raft 成员数为 1。
 + 默认值：false
 
+### `tso-update-physical-interval` <span class="version-mark">从 v6.4.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline pd-configuration-file.md:95]:
```suggestion
### `tso-update-physical-interval`
```
`tso-update-physical-interval` is introduced in https://github.com/tikv/pd/pull/3027.

File: pd-configuration-file.md (hunk 1 above)
Comment (pingyu) [reply pd-configuration-file.md]: 我保留了 ”PD **最多**提供 262144 个 TSO“。

IMO 这样会准确一些。262144 只是上限，PD 并不总是提供这么多 TSO，而跟实际请求量有关。

File: pd-configuration-file.md (hunk 2 above)
Comment (pingyu) [reply pd-configuration-file.md]: Fixed, PTAL~

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1842,21 +1840,37 @@ Raft Engine 相关的配置项。
...
-+ TiKV 会根据前一周期的使用情况，来调整时间戳的缓存数量。如果本地缓存使用率偏低，TiKV 会逐步降低缓存数量，直至等于 `renew-batch-min-size`。如果业务中经常出现突发的大流量写入，可以适当调大这个参数。注意这个参数是单个 tikv-server 的缓存大小，如果配置过大、而同时集群中 tikv-server 较多，会导致 TSO 消耗过快。
++ 时间戳单次请求的最小数量。
++ TiKV 会根据前一周期的使用情况以及 `alloc-ahead-buffer` 参数，来调整时间戳的缓存数量。如果 TSO 需求量较低，TiKV 会降低单次 TSO 请求量，直至等于 `renew-batch-min-size`。如果业务中经常出现突发的大流量写入，可以适当调大这个参数。注意这个参数是单个 tikv-server 的缓存大小，如果配置过大、而同时集群中 tikv-server 较多，会导致 TSO 消耗过快。
 + Grafana **TiKV-Raw** 面板下 **Causal timestamp** 中的 **TSO batch size** 是根据业务负载动态调整后的本地缓存数量。可以参考该监控指标值调整这个参数的大小。
 + 默认值：100
+
+### `renew-batch-max-size` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
++ 时间戳单次请求的最大数量。
++ PD 的 TSO 服务在每个物理时钟周期最多提供 262144 个 TSO，超过这个数量后 PD 会暂缓 TSO 请求的处理。这个配置用于避免 PD 的 TSO 消耗殆尽、影响其他业务的使用。如果增大这个参数，建议同时减小 PD 的 [`tso-update-physical-interval`](/pd-configuration-file.md#tso-update-physical-interval) 参数，以获得足够的 TSO 供应。
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1875]:
```suggestion
+ 在默认的一个 TSO 物理时钟更新周期内 (50ms)，PD 最多提供 262144 个 TSO，超过这个数量后 PD 会暂缓 TSO 请求的处理。这个配置用于避免 PD 的 TSO 消耗殆尽、影响其他业务的使用。如果增大这个参数，建议同时减小 PD 的 [`tso-update-physical-interval`](/pd-configuration-file.md#tso-update-physical-interval) 参数，以获得足够的 TSO 供应。
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1842,21 +1840,37 @@ Raft Engine 相关的配置项。
...
 
-> **警告：**
->
-> - API V2 是 TiKV 在 v6.1.0 中引入的实验特性，不建议在生产环境中使用。
+### `alloc-ahead-buffer` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
++ 预分配 TSO 缓存大小（以时长计算）。
++ 表示 TiKV 将按照这个参数指定的时长，预分配 TSO 缓存。TiKV 会根据前一周期的使用情况，预估并请求满足 `alloc-ahead-buffer` 时长所需要的 TSO 数量，缓存在本地。
++ 这个参数通常用于提高 TiKV API V2 (`storage.api-version = 2`) 对 PD 故障的容忍度。
++ 调大这个参数会增加 TSO 消耗，并增加 TiKV 的内存开销。为了获得足够的 TSO 供应，建议同时调小 PD 的 [`tso-update-physical-interval`](/pd-configuration-file.md#tso-update-physical-interval) 参数。
```
Comment (shichun-0415) [inline tikv-configuration-file.md:1850]:
```suggestion
+ 调大这个参数会增加 TSO 消耗，并增加 TiKV 的内存开销。为了获得足够的 TSO，建议同时调小 PD 的 [`tso-update-physical-interval`](/pd-configuration-file.md#tso-update-physical-interval) 参数。
```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1842,21 +1840,37 @@ Raft Engine 相关的配置项。
...
++ 根据测试，在 PD 主节点故障并切换到其他节点过程中，默认配置下写请求会短暂出现延迟增大和 QPS 下降（幅度约 15%）。
```
Comment (shichun-0415) [inline tikv-configuration-file.md:1851]:
```suggestion
+ 根据测试，默认配置下，当 PD 主节点由于故障切换到其节点时，写请求会短暂出现延迟增大和 QPS 下降（幅度约 15%）。
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -1842,21 +1840,37 @@ Raft Engine 相关的配置项。
...
++ 预分配 TSO 缓存大小（以时长计算）。
++ 表示 TiKV 将按照这个参数指定的时长，预分配 TSO 缓存。TiKV 会根据前一周期的使用情况，预估并请求满足 `alloc-ahead-buffer` 时长所需要的 TSO 数量，缓存在本地。
++ 这个参数通常用于提高 TiKV API V2 (`storage.api-version = 2`) 对 PD 故障的容忍度。
++ 调大这个参数会增加 TSO 消耗，并增加 TiKV 的内存开销。为了获得足够的 TSO，建议同时调小 PD 的 [`tso-update-physical-interval`](/pd-configuration-file.md#tso-update-physical-interval) 参数。
++ 根据测试，默认配置下，当 PD 主节点由于故障切换到其节点时，写请求会短暂出现延迟增大和 QPS 下降（幅度约 15%）。
++ 如果希望业务不受影响，可以尝试采用以下配置：
+    + `causal-ts.alloc-ahead-buffer = "6s"`
+    + `causal-ts.renew-batch-max-size = 65536`
+    + `causal-ts.renew-batch-min-size = 2048`
+    + 此外，在 PD 中配置 `tso-update-physical-interval = "1ms"`
```
Comment (shichun-0415) [inline tikv-configuration-file.md:1856]: 如果这个参数本身就是需要在 PD 配的，不会引起歧义，建议采用跟上面一样的风格
```suggestion
    + `tso-update-physical-interval = "1ms"`
```

File: tikv-configuration-file.md [hunk 7]
```diff
@@ -1842,21 +1840,37 @@ Raft Engine 相关的配置项。
...
-+ TiKV 会每隔 `renew-interval` 发起一次时间戳更新，并根据前一周期的使用情况，来调整时间戳的缓存数量。这个参数配置过大会导致不能及时反映最新的 TiKV 负载变化。而配置过小则会增加 PD 的负载。如果写流量剧烈变化、频繁出现时间戳耗尽、写延迟增加，可以适当调小这个参数，但需要同时关注 PD 的负载情况。
++ 更新本地缓存时间戳的周期。
++ TiKV 会每隔 `renew-interval` 发起一次时间戳更新，并根据前一周期的使用情况以及 `alloc-ahead-buffer` 参数，来调整时间戳的缓存数量。这个参数配置过大会导致不能及时反映最新的 TiKV 负载变化。而配置过小则会增加 PD 的负载。如果写流量剧烈变化、频繁出现时间戳耗尽、写延迟增加，可以适当调小这个参数，但需要同时关注 PD 的负载情况。
 + 默认值：100ms
 
 ### `renew-batch-min-size`
 
-+ 时间戳缓存的最小数量。
-+ TiKV 会根据前一周期的使用情况，来调整时间戳的缓存数量。如果本地缓存使用率偏低，TiKV 会逐步降低缓存数量，直至等于 `renew-batch-min-size`。如果业务中经常出现突发的大流量写入，可以适当调大这个参数。注意这个参数是单个 tikv-server 的缓存大小，如果配置过大、而同时集群中 tikv-server 较多，会导致 TSO 消耗过快。
++ 时间戳单次请求的最小数量。
```
Comment (shichun-0415) [inline tikv-configuration-file.md:1867]:
```suggestion
+ 单次时间戳请求的最小数量。
```

File: tikv-configuration-file.md [hunk 8]
```diff
@@ -1842,21 +1840,37 @@ Raft Engine 相关的配置项。
...
++ TiKV 会根据前一周期的使用情况以及 `alloc-ahead-buffer` 参数，来调整时间戳的缓存数量。如果 TSO 需求量较低，TiKV 会降低单次 TSO 请求量，直至等于 `renew-batch-min-size`。如果业务中经常出现突发的大流量写入，可以适当调大这个参数。注意这个参数是单个 tikv-server 的缓存大小，如果配置过大、而同时集群中 tikv-server 较多，会导致 TSO 消耗过快。
```
Comment (shichun-0415) [inline tikv-configuration-file.md:1868]:
```suggestion
+ TiKV 会根据前一周期的使用情况以及 `alloc-ahead-buffer` 参数设置，来调整时间戳的缓存数量。如果 TSO 需求量较低，TiKV 会降低单次 TSO 请求量，直至等于 `renew-batch-min-size`。如果业务中经常出现突发的大流量写入，可以适当调大这个参数。注意这个参数是单个 tikv-server 的缓存大小，如果配置过大、而同时集群中 tikv-server 较多，会导致 TSO 消耗过快。
```

File: tikv-configuration-file.md [hunk 9]
```diff
@@ -1842,21 +1840,37 @@ Raft Engine 相关的配置项。
...
 + Grafana **TiKV-Raw** 面板下 **Causal timestamp** 中的 **TSO batch size** 是根据业务负载动态调整后的本地缓存数量。可以参考该监控指标值调整这个参数的大小。
 + 默认值：100
+
+### `renew-batch-max-size` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
++ 时间戳单次请求的最大数量。
++ 在默认的一个 TSO 物理时钟更新周期内 (50ms)，PD 最多提供 262144 个 TSO，超过这个数量后 PD 会暂缓 TSO 请求的处理。这个配置用于避免 PD 的 TSO 消耗殆尽、影响其他业务的使用。如果增大这个参数，建议同时减小 PD 的 [`tso-update-physical-interval`](/pd-configuration-file.md#tso-update-physical-interval) 参数，以获得足够的 TSO 供应。
```
Comment (shichun-0415) [inline tikv-configuration-file.md:1875]:
```suggestion
+ 在默认的一个 TSO 物理时钟更新周期内 (50ms)，PD 最多提供 262144 个 TSO，超过这个数量后 PD 会暂缓 TSO 请求的处理。这个配置用于避免 PD 的 TSO 消耗殆尽、影响其他业务的使用。如果增大这个参数，建议同时减小 PD 的 [`tso-update-physical-interval`](/pd-configuration-file.md#tso-update-physical-interval) 参数，以获得足够的 TSO。
```

**Review by shichun-0415 — APPROVED** [review]
Rest LGTM

File: pd-configuration-file.md [hunk 10]
```diff
@@ -92,6 +92,14 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
 + 强制让该 PD 以一个新集群启动，且修改 raft 成员数为 1。
 + 默认值：false
 
+### `tso-update-physical-interval`
+
++ TSO 物理时钟更新周期。
++ 在默认的一个 TSO 物理时钟更新周期内 (50ms)，PD 最多提供 262144 个 TSO。如果需要更多的 TSO，可以将这个参数调小。最小值为 `1ms`。
++ 缩短这个参数会增加 PD 的 CPU 消耗。根据测试，`1ms` 相比 `50ms` 额外消耗约 0.1 个 CPU core。
```
[the remark this answers is not in the document]
Comment (pingyu) [reply pd-configuration-file.md]: `10% CPU core` 容易引起歧义
这里的 0.1 个 CPU core 是指“1 个 CPU core 的 10%”；因为机器上通常由多个 CPU core，因此“10% CPU core”容易被误解为所有 CPU core 的 10%

File: tikv-configuration-file.md (hunk 6 above)
Comment (pingyu) [reply tikv-configuration-file.md]: 其他参数是在 TiKV 配置，最后这个是在 PD 配置，如果不加以注明，会引起误会

File: pd-configuration-file.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (shichun-0415) [reply pd-configuration-file.md]: 这个中文当时看确实没问题，我是看英文的时候，发现这个表达很奇怪，如果按照当前的中文，英文大概就是 According to the test, compared with the interval of `50ms`, PD consumes about 0.1 more CPU core when the interval is `1ms`.

[the remark this answers is not in the document]
Comment (pingyu) [reply pd-configuration-file.md]: 或者改成这样吧:
PD 会额外增加约 10% 的 CPU 占用率（[CPU Usage](https://man7.org/linux/man-pages/man1/top.1.html)）

对应的英文
PD consumes about 10% more [CPU Usage](https://man7.org/linux/man-pages/man1/top.1.html)

File: pd-configuration-file.md [hunk 11]
```diff
@@ -92,6 +92,14 @@ PD 配置文件比命令行参数支持更多的选项。你可以在 [conf/conf
 + 强制让该 PD 以一个新集群启动，且修改 raft 成员数为 1。
 + 默认值：false
 
+### `tso-update-physical-interval`
+
++ TSO 物理时钟更新周期。
++ 在默认的一个 TSO 物理时钟更新周期内 (50ms)，PD 最多提供 262144 个 TSO。如果需要更多的 TSO，可以将这个参数调小。最小值为 `1ms`。
++ 缩短这个参数会增加 PD 的 CPU 消耗。根据测试，相比 `50ms` 更新周期，更新周期为 `1ms` 时，PD 会额外消耗约 0.1 CPU core。
```
Comment (Oreoxmt) [inline pd-configuration-file.md:99]:
```suggestion
+ 缩短这个参数会增加 PD 的 CPU 消耗。根据测试，相比 `50ms` 更新周期，更新周期为 `1ms` 时，PD 的 CPU 占用率 ([CPU usage](https://man7.org/linux/man-pages/man1/top.1.html)) 将增加约 10%。
```

**Pull request merged**